host/tests/dsCompositeIn_L3_Tests/dsCompositeIn_test4_VerifySignal_Callback.py

```diff
@@ -0,0 +1,119 @@
+# * following copyright and licenses apply:
+# *
+# * Copyright 2024 RDK Management
+# *
+# * Licensed under the Apache License, Version 2.0 (the "License");
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# *
+# http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import os
+import sys
+from enum import Enum, auto
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path, "../"))
+
+from dsCompositeInHelperClass import dsCompositeInHelperClass
+
+class dsCompositeIn_test4_VerifySignal_Callback(dsCompositeInHelperClass):
+    """
+    Test class to verify the signal change callback of CompositeIn device
+
+    This class uses the `dscompositeInClass` to interact with the device's compositeIn ports,
+    downloading necessary test assets, selecting
+    CompositeIn ports, and performing verification of compositeIn output.
+    """
+
+    def __init__(self):
+        """
+        Initializes the test4_VerifySignal_Callback test .
+
+        Args:
+            None.
+        """
+        self.testName  = "test4_VerifySignal_Callback"
+        super().__init__(self.testName, '1')
+
+     #TODO: Current version supports only manual verification.
+    def CheckDeviceStatus(self, manual=False, connect=False, port_type:str=0):
+        """
+        Verifies whether Composite Source Device connected or not
+
+        Args:
+            port_type (str) : CompositeIn port
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                     Defaults to other verification methods
+
+        Returns:
+            bool
+        """
+        if manual == True and connect == True:
+            return self.testUserResponse.getUserYN(f"Check if CompositeIn source is connected to {port_type} and press Y:")
+        elif manual == True and connect == False:
+            return self.testUserResponse.getUserYN(f"Check if CompositeIn source is disconnected to {port_type}/set to standby and press Y:")
+        else :
+            #TODO: Add automation verification methods
+            return False
+
+    def testFunction(self):
+        """
+        The main test function that verifies signal status of CompositeIn device.
+
+        This function:
+        - Verifies CompositeIn signal status through callbacks.
+
+        Returns:
+            bool: Final result of the test.
+        """
+
+        self.log.testStart("test4_VerifySignal_Callback", '1')
```

Comment:
test number has to updated correctly as per tests

Reply:
done